scaleway/compute_instance_helpers.go

```diff
@@ -0,0 +1,25 @@
+package scaleway
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+// getInstanceAPIWithZone returns a new instance API and the zone for a Create request
+func getInstanceAPIWithZone(d *schema.ResourceData, m interface{}) (*instance.API, utils.Zone, error) {
+	meta := m.(*Meta)
+	instanceApi := instance.NewAPI(meta.scwClient)
+
+	zone, err := getZone(d, meta)
+	return instanceApi, zone, err
+}
+
+// getInstanceAPIWithZoneAndID returns an instance API with zone and ID extracted from the state
+func getInstanceAPIWithZoneAndID(m interface{}, id string) (*instance.API, utils.Zone, string, error) {
```

Comment:
If we need another data in the future the method will be called `getInstanceAPIWithZoneAndIDAndPlopAndPlip`? :/

What do you think about something like `getInstanceAPIAndGenerics`? It could return a `struct` if we need more data in the future.

Reply:
Couldn't we store that in the meta object somewhere?

Reply:
This is a internal that we are almos certainly going to change.....
Let's pick a simple solution an update later if needed